### ISSUE TITLE
[Bug] When training a custom dataset (kitti format), I encountered the problem that AP was always 0

### ISSUE THREAD

**Issue by wisdomwxy** [open-mmlab/mmdetection3d#3117]
### Prerequisite

- [x] I have searched [Issues](https://github.com/open-mmlab/mmdetection3d/issues) and [Discussions](https://github.com/open-mmlab/mmdetection3d/discussions) but cannot get the expected help.
- [x] I have read the [FAQ documentation](https://mmdetection3d.readthedocs.io/en/latest/notes/faq.html) but cannot get the expected help.
- [x] The bug has not been fixed in the [latest version (dev-1.x)](https://github.com/open-mmlab/mmdetection3d/tree/dev-1.x) or [latest version (dev-1.0)](https://github.com/open-mmlab/mmdetection3d/tree/dev-1.0).

### Task

I'm using the official example scripts/configs for the officially supported tasks/models/datasets.

### Branch

main branch https://github.com/open-mmlab/mmdetection3d

### Environment

Hello, distinguished scholars! I had an issue with AP always being 0 while training on a custom dataset (kitti format). First, I used CVAT
Annotate the point cloud with two classes: 'Pedestrian' and 'Car' using tools/misc/ browse_datasate.py
The tool for visualization also displays correctly. The trained model is' second ', the data set has 20 point clouds (the data set is small because I want to verify the training is correct), and the training rounds are 80
Wheel. Both the gradient and loss are decreasing during training, but the validated AP values are all 0. In order to verify whether it is the problem of the environment, I used the slightly changed configuration text to train on kitti data, and verified that the AP value of the result was normal. Here's my config file:

### Reproduces the problem - code sample

_base_ = [
    '../_base_/models/second_hv_secfpn_kitti.py',
    '../_base_/schedules/cyclic-40e.py',
    '../_base_/default_runtime.py'
]

 
dataset_type = 'KittiDataset'# 数据集配置
data_root = 'data/roadway20/'
class_names = ['Pedestrian', 'Car']  # 只使用数据集中存在的两个类别
metainfo = dict(classes=class_names)
backend_args = None

point_cloud_range = [0, -40, -3, 70.4, 40, 1]

voxel_size = [0.05, 0.05, 0.1]

sparse_shape = [41, 1600, 1408]

input_modality = dict(use_lidar=True, use_camera=False)

model = dict(
    type='VoxelNet',
    data_preprocessor=dict(
        type='Det3DDataPreprocessor',
        voxel=True,
        voxel_layer=dict(
            max_num_points=32,
            point_cloud_range=point_cloud_range,
            voxel_size=voxel_size,
            max_voxels=(16000, 40000)),
        batch_first=True),
    voxel_encoder=dict(type='HardSimpleVFE'),
    middle_encoder=dict(
        type='SparseEncoder',
        in_channels=4,
        sparse_shape=sparse_shape,
        order=('conv', 'norm', 'act')),
    backbone=dict(
        type='SECOND',
        in_channels=256,
        layer_nums=[5, 5],
        layer_strides=[1, 2],
        out_channels=[128, 256]),
    neck=dict(
        type='SECONDFPN',
        in_channels=[128, 256],
        upsample_strides=[1, 2],
        out_channels=[256, 256]),
    bbox_head=dict(
        type='Anchor3DHead',
        num_classes=2,  # 修改为2个类别
        in_channels=512,
        feat_channels=512,
        use_direction_classifier=True,
        anchor_generator=dict(
            type='Anchor3DRangeGenerator',
            ranges=[
                [0, -40.0, -0.6, 70.4, 40.0, -0.6],   # Pedestrian
                [0, -40.0, -1.78, 70.4, 40.0, -1.78], # Car
            ],
            sizes=[
                [0.8, 0.6, 1.73],   # Pedestrian
                [3.9, 1.6, 1.56],   # Car
            ],
            rotations=[0, 1.57],
            reshape_out=False),
        diff_rad_by_sin=True,
        bbox_coder=dict(type='DeltaXYZWLHRBBoxCoder'),
        loss_cls=dict(
            type='mmdet.FocalLoss',
            use_sigmoid=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
        loss_bbox=dict(
            type='mmdet.SmoothL1Loss', beta=1.0 / 9.0, loss_weight=2.0),
        loss_dir=dict(
            type='mmdet.CrossEntropyLoss', use_sigmoid=False,
            loss_weight=0.2)),
    train_cfg=dict(
        assigner=[
            dict(
                type='Max3DIoUAssigner',
                iou_calculator=dict(type='BboxOverlapsNearest3D'),
                pos_iou_thr=0.1,  # 降低正样本IoU阈值
                neg_iou_thr=0.1,  # 降低负样本IoU阈值
                min_pos_iou=0.1,  # 降低最小正样本IoU
                ignore_iof_thr=-1),
            dict(
                type='Max3DIoUAssigner',
                iou_calculator=dict(type='BboxOverlapsNearest3D'),
                pos_iou_thr=0.2,  # 降低正样本IoU阈值
                neg_iou_thr=0.2,  # 降低负样本IoU阈值
                min_pos_iou=0.2,  # 降低最小正样本IoU
                ignore_iof_thr=-1),
        ],
        allowed_border=0,
        pos_weight=-1,
        debug=False),
    test_cfg=dict(
        score_thr=0.01,  # 降低得分阈值
        nms_pre=100,
        max_num=50,
        nms_thr=0.01,
        use_rotate_nms=True,
        min_bbox_size=0,
        nms_across_levels=False))

train_pipeline = [
    dict(
        type='LoadPointsFromFile',
        coord_type='LIDAR',
        load_dim=4,
        use_dim=4,
        backend_args=backend_args),
    dict(type='LoadAnnotations3D', with_bbox_3d=True, with_label_3d=True, backend_args=backend_args),
    dict(type='PointsRangeFilter', point_cloud_range=point_cloud_range),
    dict(type='ObjectRangeFilter', point_cloud_range=point_cloud_range),
    dict(type='PointShuffle'),
    dict(type='Pack3DDetInputs', keys=['points', 'gt_bboxes_3d', 'gt_labels_3d'])
]

test_pipeline = [
    dict(
        type='LoadPointsFromFile',
        coord_type='LIDAR',
        load_dim=4,
        use_dim=4,
        backend_args=backend_args),
    dict(
        type='MultiScaleFlipAug3D',
        img_scale=(1333, 800),
        pts_scale_ratio=1,
        flip=False,
        transforms=[
            dict(
                type='GlobalRotScaleTrans',
                rot_range=[0, 0],
                scale_ratio_range=[1., 1.],
                translation_std=[0, 0, 0]),
            dict(type='RandomFlip3D'),
            dict(type='PointsRangeFilter', point_cloud_range=point_cloud_range),
        ]),
    dict(type='Pack3DDetInputs', keys=['points'])
]

train_dataloader = dict(
    batch_size=6,
    num_workers=4,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    dataset=dict(
        type='RepeatDataset',
        times=2,
        dataset=dict(
            type=dataset_type,
            data_root=data_root,
            ann_file='roadway20_infos_train.pkl',  # 使用原始文件
            data_prefix=dict(pts='training/velodyne'),  # 指定点云文件路径
            pipeline=train_pipeline,
            modality=input_modality,
            test_mode=False,
            metainfo=metainfo,
            box_type_3d='LiDAR',
            backend_args=backend_args)))

val_dataloader = dict(
    batch_size=1,
    num_workers=1,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        ann_file='roadway20_infos_val.pkl',  # 使用原始文件
        data_prefix=dict(pts='training/velodyne'),  # 指定点云文件路径
        pipeline=test_pipeline,
        modality=input_modality,
        test_mode=True,
        metainfo=metainfo,
        box_type_3d='LiDAR',
        backend_args=backend_args))

test_dataloader = dict(
    batch_size=1,
    num_workers=1,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        ann_file='roadway20_infos_val.pkl',  # 使用验证集作为测试集，因为测试集没有标注
        data_prefix=dict(pts='training/velodyne'),  # 指定点云文件路径
        pipeline=test_pipeline,
        modality=input_modality,
        test_mode=True,
        metainfo=metainfo,
        box_type_3d='LiDAR',
        backend_args=backend_args))

val_evaluator = dict(
    type='KittiMetric',
    ann_file=data_root + 'roadway20_infos_val.pkl',
    metric='bbox',
    backend_args=backend_args)

test_evaluator = val_evaluator

train_cfg = dict(by_epoch=True, max_epochs=80, val_interval=5)

default_hooks = dict(
    timer=dict(type='IterTimerHook'),
    logger=dict(type='LoggerHook', interval=50),
    param_scheduler=dict(type='ParamSchedulerHook'),
    checkpoint=dict(type='CheckpointHook', interval=10),  # 每10个epoch保存一次检查点
    sampler_seed=dict(type='DistSamplerSeedHook'),
    visualization=dict(type='Det3DVisualizationHook'))

optim_wrapper = dict(
    type='OptimWrapper',
    optimizer=dict(type='AdamW', lr=0.001, weight_decay=0.01),  # 降低学习率
    clip_grad=dict(max_norm=10, norm_type=2))

param_scheduler = [
    dict(
        type='CosineAnnealingLR',
        T_max=80,
        eta_min=1e-5,
        begin=0,
        end=80,
        by_epoch=True,
        convert_to_iter_based=True)
]

env_cfg = dict(
    cudnn_benchmark=False,
    mp_cfg=dict(mp_start_method='fork', opencv_num_threads=0),
    dist_cfg=dict(backend='nccl'))

vis_backends = [dict(type='LocalVisBackend')]
visualizer = dict(
    type='Det3DLocalVisualizer', vis_backends=vis_backends, name='visualizer')

log_level = 'INFO'
log_processor = dict(type='LogProcessor', window_size=50, by_epoch=True)

load_from = None
resume = False

randomness = dict(seed=0, deterministic=False)

auto_scale_lr = dict(enable=True, base_batch_size=48)

### Reproduces the problem - command or script

python tool/train.py

### Reproduces the problem - error message

nothing

**Comment by Xrenya** [comment]
Did you confirm that you dataset has the same structure for annotation, like bounding box orientation?
